pkg/sql/schema_resolver.go

```diff
@@ -590,6 +590,18 @@ func (sr *schemaResolver) ResolveFunctionByOID(
 	return fnName, ret, nil
 }
 
+// FunctionDesc returns the descriptor for the function with the given OID.
+func (sr *schemaResolver) FunctionDesc(
+	ctx context.Context, oid oid.Oid,
+) (catalog.FunctionDescriptor, error) {
+	if !funcdesc.IsOIDUserDefinedFunc(oid) {
+		return nil, errors.Wrapf(tree.ErrFunctionUndefined, "function %d not user-defined", oid)
+	}
+	g := sr.byIDGetterBuilder().WithoutNonPublic().WithoutOtherParent(sr.typeResolutionDbID).Get()
+	descID := funcdesc.UserDefinedFunctionOIDToID(oid)
+	return g.Function(ctx, descID)
```

Comment:
@chengxiong-ruan is this function expensive? We'll be calling it everytime a UDF invocation is encountered by the optimizer.

Reply:
it's calling `getDescriptorsByID` internally, it will hit cache of the leasing layer. Should be fine.